content/learn/book/graphics/rendering-internals/technical-summary.md

```diff
@@ -0,0 +1,852 @@
++++
+title = "Technical Summary"
+insert_anchor_links = "right"
+[extra]
+weight = 1
+status = 'hidden'
++++
+
+## Introduction
+This document provides a technical summary of Bevy's rendering architecture as
+of the 0.13 release.
+
+Bevy's rendering code is largely concentrated in three loosely-coupled crates:
+`bevy_render`, `bevy_core_pipeline`, and `bevy_pbr`. The foundation is
+`bevy_render`, which manages how low-level rendering tasks are scheduled and
+executed. Above this sits `bevy_core_pipeline`, which defines the individual
+steps of rendering each frame. Above both is `bevy_pbr` which adds a material
+abstraction and uses it to provide physically-based shaders.
+
+These three crates sit roughly in "upper-middle" of bevy's default stack and do
+not fully encompass all of bevy's rendering capabilities. Although we will not
+cover them here, both `bevy_ui` and `bevy_sprite` also hook-into and interact
+with rendering.
+
+We will begin with the lowest levels of the stack and progress upwards towards
+high-level concepts like materials. But before diving in, lets take a look at
+the two big dependencies underpinning the rest of the rendering stack: `wgpu`
+and `bevy_ecs`.
+
+## `wgpu` Preliminaries
+This is a quick overview of some important aspects of wgpu. Those experienced
+with wgpu can skip this section. Those unfamiliar with wgpu should check out
+[the wgpu book][wgpu book] and [the WebGPU spec][WebGPU spec].
+
+`wgpu` is a rust package providing a cross platform implementation `WebGPU`, a
+modern graphics api. It strikes a nice balance between the ergonomics and power,
+and can run pretty much anywhere.
+
+[wgpu book]: https://sotrh.github.io/learn-wgpu/
+[WebGPU spec]: https://gpuweb.github.io/gpuweb/ 
+
+### Layers of Abstraction
+`wgpu` is a low level graphics library, but because it is designed to work
+across a wide beret of platforms there are many layers of abstraction between it
+and the graphics hardware. Going from highest to lowest level:
+1. The `WebGPU` api provided by `wgpu`,
```

Comment:
This is my personal preference but I think we should reverse this list. For one, it's written from highest to lowest level, but the numbers go from lowest to highest. Also the abstraction feels less "obvious" in this order. As in, I think it makes more sense to say "start from the physical GPU, and slowly make it less and less specific". Maybe that's just me.

If you change the, make sure to renumber the references to the layers in following paragraphs!